src/main/java/org/example/service/EmployeeService.java

```diff
@@ -0,0 +1,102 @@
+package org.example.service;
+
+import org.example.DTO.EmployeeDTO;
+import org.example.Exception.FieldCannotBeNullException;
+import org.example.model.DepartmentModel;
+import org.example.model.EmployeeModel;
+import org.example.model.TeamModel;
+import org.example.repository.EmployeeRepository;
+import org.example.Mapper.EmployeeMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import jakarta.validation.Valid;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class EmployeeService {
+
+    @Autowired
+    private EmployeeRepository employeeRepository;
+
+    @Autowired
+    private DepartmentService departmentService;
+
+    @Autowired
+    private TeamService teamService;
+
+    public EmployeeDTO saveEmployee(@Valid EmployeeDTO employeeDTO) {
+        // Validate employee fields
+        validateEmployeeFields(employeeDTO);
+
+        // Convert DTO to model using the mapper
+        EmployeeModel employee = EmployeeMapper.INSTANCE.toModel(employeeDTO);
+
+        // Set department if departmentId is provided
+        if (employeeDTO.getDepartmentId() != null) {
+            Optional<DepartmentModel> department = departmentService.getDepartmentById(employeeDTO.getDepartmentId());
+            employee.setDepartment(department.orElse(null));
```

Comment:
why orElse null if no department then throw not found but handle it in the department service ofcourse